public/src/pages/main/Userprofile.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { FormEvent, useEffect, useState } from 'react';
+import { Redirect } from 'react-router';
+import { getUserData, putUserData } from 'src/api/httpClient';
+import { UserData } from 'src/interfaces/User';
+import { RoutePath } from 'src/router/RoutePath';
+import AuthLayout from '../auth/AuthLayout';
+
+const defaultUserData: UserData = {
+  email: '',
+  firstName: '',
+  lastName: '',
+  isAdmin: false
+};
+
+export const Userprofile = () => {
+  const email: string | null = sessionStorage.getItem('email');
+  const [user, setUser] = useState<UserData>(defaultUserData);
+
+  const onInput = ({ target }: { target: EventTarget | null }) => {
+    const { name, value } = target as HTMLInputElement;
+    setUser({ ...user, [name]: value });
+  };
+
+  useEffect(() => {
+    email ? getUserData(email).then((data) => setUser(data)) : '';
+  }, []);
+
+  const sendUserData = (e: FormEvent) => {
+    e.preventDefault();
+    putUserData(user);
+    sessionStorage.setItem('userName', `${user.firstName} ${user.lastName}`);
+    setTimeout(() => {
+      window.location.href = RoutePath.Home;
+    }, 1500);
```

Comment:
It looks like a dirty hack) Why don't await that `putUserData()` will be resolved?